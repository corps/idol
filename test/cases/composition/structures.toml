[AliasStructFits]
is_a = ["primitive_widens.IntAlias[]", "int[]"]

[ContraOtherStruct]
is_a = ["primitive_widens.IntAlias[]", "int{}"]
variance = "Contravariant"

[IntoAny]
is_a = ["primitive_widens.IntAlias[]", "any"]

[IntoAnyStruct]
is_a = ["primitive_widens.IntAlias[]", "any[]"]

[IntoContraAnyStruct]
is_a = ["primitive_widens.IntAlias{}", "any{}"]
variance = "Contravariant"
