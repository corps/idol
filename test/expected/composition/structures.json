{
  "AliasStructFits": {
    "enum": [],
    "fields": {},
    "is_a": [
      "primitive_widens.IntAlias[]"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "ContraOtherStruct": {
    "enum": [],
    "fields": {},
    "is_a": [
      "any"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "IntoAny": {
    "enum": [],
    "fields": {},
    "is_a": [
      "primitive_widens.IntAlias[]"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "IntoAnyStruct": {
    "enum": [],
    "fields": {},
    "is_a": [
      "primitive_widens.IntAlias[]"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "IntoContraAnyStruct": {
    "enum": [],
    "fields": {},
    "is_a": [
      "any[]"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  }
}
